### PR TITLE
Update to alpine 3.16 & etcdctl 3.5.4

### PR DESCRIPTION
**Pull request by amdprophet**
Signed-off-by: Justin Kolberg <amd.prophet@gmail.com>

Closes https://github.com/sensu/sensu-release/issues/54.

### CODE REVIEW

**Review by echlebek — APPROVED** [review]

File: dockerfiles/alpine/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,12 @@
-FROM alpine:3.12 as etcdctl-fetch
+ARG ALPINE_VERSION=3.16
+FROM alpine:$ALPINE_VERSION as etcdctl-fetch
 ARG TARGETPLATFORM
 ARG BUILDPLATFORM
+ARG ALPINE_VERSION
```
Comment (ccressent) [inline dockerfiles/alpine/Dockerfile:5]: I'm curious, why this again? Isn't line 1 enough?

**Comment by ccressent** [comment]
Nitpick: in the shellscript lines, you should probably use `${VARIABLE}` when in the middle of some string, just in case someone less versed in shell comes after you, edits the string and potentially breaks the variable reference :)

File: dockerfiles/alpine/Dockerfile (hunk 1 above)
Comment (amdprophet) [reply dockerfiles/alpine/Dockerfile:5]: This is due to it being a multi-stage build. It allows it to be passed through each stage.

**Comment by amdprophet** [comment]
> Nitpick: in the shellscript lines, you should probably use `${VARIABLE}` when in the middle of some string, just in case someone less versed in shell comes after you, edits the string and potentially breaks the variable reference :)

I had thought that curly braces weren't supported for ARG substitution in Dockerfiles but it appears to work. I've updated the Dockerfile to use them now.

File: dockerfiles/alpine/Dockerfile (hunk 1 above)
Comment (ccressent) [reply dockerfiles/alpine/Dockerfile:5]: But doesn't line 1 `ARG ALPINE_VERSION=xxx` achieve the same (with a default value of `xxx`)?

**Review by ccressent — APPROVED** [review]

**Pull request merged**